HW2/P2/P2.txt

```diff
@@ -0,0 +1,60 @@
+# Solutions for HW2 - P2
+
+Uncorrelated data storage - 
+
+In this case, the serial implementation had the best performance. The serialization involved with reading from memory meant that the multi-thread approach couldnt be exploited for a problem of this size. 
+Interestingly, the medium grained approach had the worst performance irrespective of the coarseness of the lock. And as can be seen from the data below and the associated plot, on average the coarser the locks, the worse the performance. There are however, local peaks at size 15 and size 50 that I can't explain.
+
+The best choice of N here would be 5 based on the empirical results below. But its unclear if that is the best approach in the case of uncorrelated data storage as both serial and fine grained perform much better.
+
```

Comment:
Good visualization of your results! I wonder if you would have seen additional peaks if you sampled more points between N = 100 and N = 200.